lib/bulk_replace_file.rb

```diff
@@ -0,0 +1,77 @@
+require "util"
+require "diffy"
+
+# rubocop:disable Layout/MultilineOperationIndentation, Layout/SpaceInsideParens, Style/InverseMethods
+def filter_matches?(repo_name, if_any_exist, if_all_exist, unless_any_exist, unless_all_exist)
+  predicate = ->(path) { repo_contains_file? repo_name, path }
+  (    if_any_exist.empty? ||      if_any_exist.any?(&predicate)) &&
+  (    if_all_exist.empty? ||      if_all_exist.all?(&predicate)) &&
+  (unless_any_exist.empty? || !unless_any_exist.any?(&predicate)) &&
+  (unless_all_exist.empty? || !unless_all_exist.all?(&predicate))
+end
+# rubocop:enable Layout/MultilineOperationIndentation, Layout/SpaceInsideParens, Style/InverseMethods
+
+def diff(string1, string2)
+  Diffy::Diff.new("#{string1}\n", "#{string2}\n").to_s :color
+end
+
+def bulk_replace_file(github_token:, file_path:, old_content:, new_content:, branch:, pr_title:, pr_description:, if_any_exist:, if_all_exist:, unless_any_exist:, unless_all_exist:)
+  Octokit.access_token = github_token
+
+  quit_requested = false
+  Signal.trap("INT") do
+    print "Terminating..."
+    quit_requested = true
+  end
+
+  num_index_columns = govuk_repos.count.to_s.length
+  num_name_columns = govuk_repos.map(&:length).max
+  govuk_repos.each.with_index(1) do |repo_name, i|
+    exit 130 if quit_requested
+
+    print "[#{i.to_s.rjust(num_index_columns)}/#{govuk_repos.count}] #{repo_name.ljust(num_name_columns)} "
+
+    repo = begin
+      Octokit.repo(repo_name)
+    rescue Octokit::NotFound
+      puts "❌ repo doesn't exist (or we don't have permission)"
+      next
+    end
+
+    existing_file = get_file_contents(repo_name, file_path)
+
+    if existing_file.nil?
+      puts "⏭  file not found"
+    else
+      existing_file_content = Base64.decode64(existing_file.content)
+      if !existing_file_content.include?(old_content)
+        puts "⏭  content not found in file"
+      elsif repo_has_branch?(repo_name, branch)
+        puts "⏭  branch \"#{branch}\" already exists"
+      elsif !filter_matches?(repo_name, if_any_exist, if_all_exist, unless_any_exist, unless_all_exist)
+        puts "⏭  filters don't match"
+      else
+        puts "You are about to make the following changes:"
+        puts "-------------------------------------------"
+        puts diff(old_content, new_content)
+        puts "-------------------------------------------"
+        printf "\e[31mPress 'y' to continue: \e[0m"
```

Comment:
This is a good idea, I wonder if we should add a prompt to the other script as well 🤔